props/saber_fett263_buttons.h

```diff
@@ -293,248 +659,3857 @@ class SaberFett263Buttons : public virtual PropBase {
+          SaberBase::Lockup()) {
+#ifdef FETT263_CLASH_STRENGTH_SOUND
+        clash_type_ = CLASH_LOCKUP_END;
+        RealClashSound();
+#else
+        SaberBase::DoEndLockup();
+        SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+#endif
+        auto_lockup_on_ = false;
+      }
+      if (auto_melt_on_ &&
+          !swinging_ &&
+          fusor.swing_speed() > 60 &&
+          millis() - clash_impact_millis_ > saved_gesture_control.lockupdelay &&
+          SaberBase::Lockup()) {
+        SaberBase::DoEndLockup();
+        SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+        auto_melt_on_ = false;
+      }
+      // EVENT_PUSH
+      if (fabs(mss.x) < 3.0 &&
+          mss.y * mss.y + mss.z * mss.z > 100 &&
+          fusor.swing_speed() < 20 &&
+          fabs(fusor.gyro().x) < 5) {
+        if (millis() - push_begin_millis_ > saved_gesture_control.forcepushlen) {
+          // Checking for Clash at end of movement
+          clash_type_ = CLASH_CHECK;
+          push_begin_millis_ = millis();
+          clash_impact_millis_ = millis();
+        } 
+      } else {
+        push_begin_millis_ = millis();
+      }
+      
+    } else {
+      // EVENT_SWING - Swing On gesture control to allow fine tuning of speed needed to ignite
+      if (millis() - saber_off_time_ < MOTION_TIMEOUT) {
+        SaberBase::RequestMotion();
+        if (swinging_ && fusor.swing_speed() < 90) {
+          swinging_ = false;
+        }
+        if (!swinging_ && fusor.swing_speed() > saved_gesture_control.swingonspeed) {
+          swinging_ = true;
+          Event(BUTTON_NONE, EVENT_SWING);
+        }
+      }
+      // EVENT_THRUST
+      if (mss.y * mss.y + mss.z * mss.z < 16.0 &&
+          mss.x > 14  &&
+          fusor.swing_speed() < 150) {
+        if (millis() - thrust_begin_millis_ > 15) {
+          Event(BUTTON_NONE, EVENT_THRUST);
+          thrust_begin_millis_ = millis();
+        } 
+      } else {
+        thrust_begin_millis_ = millis();
+      }
+    }
+    if (menu_ || color_mode_ == CC_COLOR_LIST) {
+      DetectMenuTurn();
+    }
+    if (track_mode_ != PLAYBACK_OFF) {
+      TrackPlayer();
+    }
+  }
+
+  // Fett263 Track Player
+  enum TrackMode {
+    PLAYBACK_OFF,
+    PLAYBACK_LOOP,
+    PLAYBACK_ROTATE,
+    PLAYBACK_RANDOM,
+  };
+
+  void TrackPlayer() {
+    if (!track_player_) {
+#ifdef ENABLE_AUDIO
+      if (track_num_ == -1 && track_mode_ == PLAYBACK_LOOP) {
+        StartOrStopTrack();
+      } else {
+        if (track_num_ > num_tracks_) track_num_ = 0;
+        switch (track_mode_) {
+          case PLAYBACK_ROTATE:
+            track_num_ += 1;
+            break;
+          case PLAYBACK_RANDOM:
+            track_num_ = rand() % num_tracks_;
+            if (track_num_ < 0) track_num_ = num_tracks_ - 1;
+            break;
+          default:
+            break;
+        }
+        char playtrack[128];
+        RunCommandAndGetSingleLine("list_current_tracks", nullptr, track_num_, playtrack, sizeof(playtrack));
```

Comment:
If you have lots of tracks, this will cause audio glitches if you run it while the saber is on.
A V3 will handle a lot more though.
The reason for the "Effect" class is to avoid having to list files while the saber is running.

Reply:
I haven't experienced in testing but only have about 20 or so tracks per font.  How would we make tracks into an Effect if they're not name sequentially?  Almost every font I have just has the name of the song as the track name, so not sure how we'd handle?

Reply:
It's fairly complicated to do this right, although I'm not entirely sure how big of a problem it is. (Meaning; I don't know how many files you would need before you get an audio glitch.)

Making a system that can support a LOT of files is complicated, because even if we could figure out which file name to use without scanning the directory, opening the file will need to scan the directory in order to find the file anyways.

Do you see any audio underflows in the serial monitor when using this function?

Reply:
Just did a quick test and there were 2 Audio underflows at the time the next track was starting, there was a really, really subtle audio glitch for a split second and I only notice because I was listening for it.  Not sure how to handle, if it's a trade for the functionality of the Track Player and a split second glitch in the audio between tracks I'm ok with it but if it's something that can be improved then I'm all for it.  

channels: 1 rate: 44100 bits: 16
Audio underflows: 17
channels: 1 rate: 44100 bits: 16
Audio underflows: 12